### PR TITLE
Welcome Page

### LINKED ISSUE #15: Welcome Page
When a user visits the root path they should be on the welcome page which includes:

- [x] Welcome message
- [x] Brief description of the application
- [x] Button to Log in
- [x] Link to Registration

**Details:** Implement basic auth in the application allowing a user to log in with an email and password. The password should be stored in the database using [bcrypt](https://github.com/codahale/bcrypt-ruby).

### PR DESCRIPTION
**Pull request by LHJE**
Closes #15 

This pull request will add:
- Spec for the Welcome Show Page
- Spec for a User Model
- Spec for a User Dashboard (which will likely need tests in the future, though I've left blank for now)  

- Updated Gems: bcrypt, shoulda-matchers
- Updated Rails helper for should-matchers
- Updated routes for '/', '/register', '/dashboard'  

- User Model
- User Dashboard Model  

- Welcome Controller
- User::Dashboard Controller  

- Welcome index view  

- User Migration
- User Seed

Notes:
- Rspec won't fully run because it needs the RegisterController, which is what @PhilipDeFraties is currently working on for Issue 14.  Once that is in place, along with an index method for RegisterController and a view page for Register, then Rspec should work, as the only test that is failing is one that is dependant upon all that.
- It appears the Travis thing is failing both it's checks, which I would love to review with you because I've got no clue what that means.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  validates_presence_of :name,
+                        :email
+
```
Comment (LHJE) [inline app/models/user.rb:6]: We could maybe remove this extra space

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,10 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  get '/', to: 'welcome#index'
+
+  get '/register', to: 'register#index'
+
+  post '/dashboard', to: 'user/dashboard#index'
+
```
Comment (LHJE) [inline config/routes.rb:9]: I've left this spaced out for clarity, so you can see what all is new.

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,10 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  get '/', to: 'welcome#index'
+
+  get '/register', to: 'register#index'
```
Comment (LHJE) [inline config/routes.rb:6]: We can change this to '/registration' or whatever you end up calling yours.  I just needed a placeholder

File: spec/features/welcome/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    it "I can click link to register, and am taken to the register page" do
+      click_link "Register"
+
+      expect(current_path).to eq("/register")
+    end
+
+  end
+end
+
+# Details: Implement basic auth in the application allowing a user to log in with an email and password. The password should be stored in the database using bcrypt.
```
Comment (LHJE) [inline spec/features/welcome/show_spec.rb:46]: I've left this in as a reminder for us to double check that this is working properly.

File: app/models/user.rb (hunk 1 above)
Comment (LHJE) [reply app/models/user.rb:6]: I went through and took that out.

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe User do
+  describe 'Validations' do
+    it {should validate_presence_of :name}
+    it {should validate_presence_of :email}
+    it {should validate_uniqueness_of :email}
+  end
```
Comment (PhilipDeFraties) [inline spec/models/user_spec.rb:8]: Need a password validation

**Review by PhilipDeFraties — APPROVED** [review]
Cool pr

**Pull request merged**